### ISSUE TITLE
Beat GATK3 performance (CPU and memory) of HaplotypeCaller

### ISSUE THREAD

**Issue by akiezun** [broadinstitute/gatk#1607]
On a whole genome and exome, the goal is to beat GATK3 performance and memory usage

**Comment by akiezun** [comment]
@droazen  let me know when this is unblocked.

**Comment by akiezun** [comment]
The GATK3 v3.5-0-g36282e4 commandlines and numbers:

running on gsa6 (has AVX)

GVCF mode
```
time java -jar /humgen/gsa-hpprojects/GATK/bin/current/GenomeAnalysisTK.jar -T HaplotypeCaller -I src/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam -R src/test/resources/large/human_g1k_v37.20.21.fasta -ERC GVCF --out a.gatk3.g.vcf
...
real	3m48.076s
user	8m45.049s
```

VCF mode
```
time java -jar /humgen/gsa-hpprojects/GATK/bin/current/GenomeAnalysisTK.jar -T HaplotypeCaller -I src/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam -R src/test/resources/large/human_g1k_v37.20.21.fasta --out a.gatk3.vcf
...
real	1m52.247s
user	2m32.789s
```

**Comment by akiezun** [comment]
Update 20160502 
GATK4 Version:09f6b60-SNAPSHOT
commit 09f6b602a175a729e07d710d112e878332876390ESC

running on gsa6

GVCF mode
```
time ./gatk-launch HaplotypeCaller -I src/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam -R src/test/resources/large/human_g1k_v37.20.21.fasta -O a.g.vcf -ERC GVCF
...
real	2m12.702s
user	4m0.566s
```

VCF mode
```
time ./gatk-launch HaplotypeCaller -I src/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam -R src/test/resources/large/human_g1k_v37.20.21.fasta -O a.vcf
...
real	1m48.297s
user	3m7.960s
```

Summary so far: 

1. GATK4 wins on GVCF mode by almost 2x over gatk3.5
2. GATK4 does not yet win on the VCF mode (in particular 'user' time is actually longer) - possibly due to lack of downsampling

**Comment by akiezun** [comment]
closing this in favor of https://github.com/broadinstitute/gatk/issues/1800 and https://github.com/broadinstitute/gatk/issues/1799 (GVCF is much higher priority because that's what we run in production)
